## Q1. What are the three measures of central tendency?

Mean, Median, and Mode

## Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset?

mean is the sum of all the values of the dataset divided by the number of observations. Median is the middle value of the dataset after arranging the dataset in a ascending or descending order. Mode is the most frequently occurring value in the dataset.

The Mean is affected by outliers in the dataset. However, median is not heavily affected by the outliers. Therefore, depending on the dataset, either mean or median can be taken to measure the central tendency of a dataset. Since mode represents the most common value in the dataset, it can be used to describe the central tendency of a dataset with categorical or nominal data.

## Q3. Measure the three measures of central tendency for the given height data:
#### [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [4]:
lst = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

import numpy as np
from scipy import stats

mean = np.mean(lst)

median = np.median(lst)

mode = stats.mode(lst)

print("Mean is {}, Median is {}, and Mode is {}".format(mean, median, int(mode.mode)))

Mean is 177.01875, Median is 177.0, and Mode is 177


/tmp/ipykernel_147/2949735012.py:10: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  mode = stats.mode(lst)


## Q4. Find the standard deviation for the given data:
### [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [6]:
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

std = np.std(data)

print("Standard deviation is {}".format(std))

Standard deviation is 1.7885814036548633


## Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.

Measures of despersion such as range, variance, and standard deviation are used to describe the variability of the dataset. The provide how far data points are away from the central tendency of measures ,and how spread out the values in the dataset are from each other.

1. range - it gives the difference between maximum and minimum values.
2. variance - it measures how much the individual values in a dataset differ from the mean.
3. standard deviation - it measures how much the values in a dataset are spread out from the mean.

In [12]:
## Range
d1 = [i for i in range(40)]
min_d1 = min(d1)
max_d1 = max(d1)

print("Range of d1 is {} - {}".format(min_d1,max_d1))

Range of d1 is 0 - 39


In [7]:
np.random.seed(123)
d1 = np.random.normal(loc=50, scale=20, size=100)
d2 = np.random.normal(loc=50, scale=5, size=100)

In [14]:
var_d1 = np.var(d1)
var_d2 = np.var(d2)

std_d1 = np.std(d1)
std_d2 = np.std(d2)

print("Variance of d1 {}, variance of d2 {}".format(var_d1,var_d2))
print("Standard deviation of d1 {}, standard deviation of d2 {}".format(std_d1,std_d2))

Variance of d1 133.25, variance of d2 23.525099114360838
Standard deviation of d1 11.543396380615196, standard deviation of d2 4.850267942532746


In the above example, variance of D1 is higher the that of d2. It can be concluded that values in the d1 are spread out over a wide range of values than d2.

At the same time, std of d1 and d2 are 11.54 and 4.85 respectively. It indicates that data points of d1 are spread out from the mean about 11.54 on average, while data points of d2 are spread out from the mean about 4.85 on average.

## Q6. What is a Venn diagram?

A Venn diagram is a visual representation of the relationships between sets of data. It consists of overlapping circles or ellipses, each representing a set, with the area of overlap representing the intersection of the sets.

## Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:

In [17]:
A = {2,3,4,5,6,7} 
B = {0,2,6,8,10}

### (i) A ^ B

In [20]:
A.intersection(B)

{2, 6}

### (ii) A ⋃ B

In [21]:
A.union(B)

{0, 2, 3, 4, 5, 6, 7, 8, 10}

## Q8. What do you understand about skewness in data?

Skewness refers to the degree of asymmetry of a distribution. A distribution is said to be skewed if it is not symmetric.
Skewness can be positive, negative, or zero. A positive skewness indicates that the tail of the distribution is longer on the right side, while a negative skewness indicates that the tail is longer on the left side. Zero skewness indicates that the distribution is symmetric.

## Q9. If a data is right skewed then what will be the position of median with respect to mean?

Median will be left site to the mean or mean will be greater than the median.

## Q10. Explain the difference between covariance and correlation. How are these measures used in statistical analysis?

Covariance measures how much two variables change together. Covariance can take on both positive and negative values, with positive values indicating a positive relationship between the variables and negative values indicating a negative relationship. However, covariance by itself does not provide information on the strength or direction of the relationship between the variables.

Correlation, on the other hand, measures the strength and direction of the linear relationship between two variables. It ranges from -1 to 1, with a correlation coefficient of 1 indicating a perfect positive linear relationship, a correlation coefficient of -1 indicating a perfect negative linear relationship, and a correlation coefficient of 0 indicating no linear relationship between the variables.

In statistical analysis, both covariance and correlation are used to assess the relationship between two variables.

## Q11. What is the formula for calculating the sample mean? Provide an example calculation for a dataset.

sample mean is calcuated by sum of all values in the dataset divided by number of dataset.

let's consider the dataset,

data = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49]

length of the dataset is 50. 

so, mean = sum of all the dataset / length of the datast

In [22]:
data = [i for i in range(50)]

In [27]:
print("Mean of data is {}".format(np.mean(data)))

Mean of data is 24.5


## Q12. For a normal distribution data what is the relationship between its measure of central tendency?

Normal distribution is symmetric. Therefore, mean, median, and mode will be the same, and are located at the central of the distribution.

## Q13. How is covariance different from correlation?

## Q13. How is covariance different from correlation?

Though both covariance and correlation are used to measure the correlation between two variables. However,  covariance does not have a standardized scale, and its magnitude is difficult to interpret. Correlation, on the other hand, has a scale between -1 to +1. So, correlation is more interpretable than covariance.

## Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

Outliers affects central tendency and dispersion. 

mean - outliers can pull the mean away from the typical position, making the distribution skewed. 

median - median is also get affected by outliers. However, it less affected compared to mean.

mode - The mode may also be affected if the outlier value is the most frequently occurring value in the dataset.

The variance and standard deviation are also influenced by outliers since they are based on the distance of each data point from the mean. An outlier that is far from the mean can significantly increase the variance and standard deviation of the dataset.

In [28]:
data_without_outliers = [1, 5, 10, 15, 20, 25]
data_with_outliers = [1, 5, 10, 15, 20, 25, 100]

In [47]:
mean_data_without_outliers = np.mean(data_without_outliers)
median_data_without_outliers = np.median(data_without_outliers)
mode_data_without_outliers = stats.mode(data_without_outliers).mode
var_data_without_outliers = np.var(data_without_outliers)
std_data_without_outliers = np.std(data_without_outliers)

mean_data_with_outliers = np.mean(data_with_outliers)
median_data_with_outliers = np.median(data_with_outliers)
mode_data_with_outliers = stats.mode(data_with_outliers).mode
var_data_with_outliers = np.var(data_with_outliers)
std_data_with_outliers = np.std(data_with_outliers)

/tmp/ipykernel_147/4257562873.py:3: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  mode_data_without_outliers = stats.mode(data_without_outliers).mode
/tmp/ipykernel_147/4257562873.py:9: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  mode_data_with_outliers = stats.

In [48]:
print("mean of data_with_outliers : {}, data_without_outliers : {}".format(mean_data_with_outliers,mean_data_without_outliers))
print("median of data_with_outliers : {}, data_without_outliers : {}".format(median_data_with_outliers,median_data_without_outliers))
print("mode of data_with_outliers : {}, data_without_outliers : {}".format(mode_data_with_outliers,mode_data_without_outliers))
print("variance of data_with_outliers : {}, data_without_outliers : {}".format(var_data_with_outliers,var_data_without_outliers))
print("standard deviation of data_with_outliers : {}, data_without_outliers : {}".format(std_data_with_outliers,std_data_without_outliers))

mean of data_with_outliers : 25.142857142857142, data_without_outliers : 12.666666666666666
median of data_with_outliers : 15.0, data_without_outliers : 12.5
mode of data_with_outliers : [1], data_without_outliers : [1]
variance of data_with_outliers : 992.9795918367348, data_without_outliers : 68.8888888888889
standard deviation of data_with_outliers : 31.51157869477083, data_without_outliers : 8.299933065325822


**In the above example, it is clear that outliers affected both the measures of central tendency and dispersion. However, median is not heavily affected.**